src/core/src/runtime/tensor.cpp

```diff
@@ -152,14 +153,10 @@ ov::Shape calc_static_shape_for_file(const std::filesystem::path& file_name,
                     " bytes");
     auto elements_to_read = file_size_to_read * 8 / element_type.bitwidth();
 
-    auto new_dimension_size = elements_to_read / slice_size;
-    OPENVINO_ASSERT(new_dimension_size * slice_size == elements_to_read,
-                    "Cannot fit file size into requested PartialShape");
-
-    OPENVINO_ASSERT(dynamic_dimension.compatible(new_dimension_size),
-                    "Cannot fit file size into requested PartialShape");
+    auto new_dimension = ov::Dimension(elements_to_read) / slice_size;
+    OPENVINO_ASSERT(dynamic_dimension.compatible(new_dimension), "Cannot fit file size into requested PartialShape");
 
-    dynamic_dimension = Dimension(new_dimension_size);
+    dynamic_dimension = new_dimension;
```

Comment:
```suggestion
    dynamic_dimension = std::move(new_dimension);
```
It is big chance Coverity will report issue copy instead move also

Reply:
ok. changed for our Coverity friend :)